include/envoy/router/http_route_manager.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/event/dispatcher.h"
+#include "envoy/init/init.h"
+#include "envoy/json/json_object.h"
+#include "envoy/local_info/local_info.h"
+#include "envoy/router/rds.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/stats/stats.h"
+#include "envoy/thread_local/thread_local.h"
+#include "envoy/upstream/cluster_manager.h"
+
+namespace Envoy {
+namespace Router {
+
+/**
+ * The HttpRouteManager exposes the ability to get a RouteConfigProvider. This interface is exposed
+ * to the Server's FactoryContext in order to allow HttpConnMans to get RouteConfigProviders.
+ */
+class HttpRouteManager {
+public:
+  virtual ~HttpRouteManager() {}
+
+  /**
+   * get a Router::RouteConfigProviderSharedPtr. Ownership of the RouteConfigProvider is shared by
+   * all the ConnectionManagers who own a Router::RouteConfigProviderSharedPtr. The HttpRouteManager
+   * holds weak_ptrs to the RouteConfigProviders. Clean up of the weak ptrs happen from the
+   * destructor of the RouteConfigProvider. This function creates a RouteConfigProvider is there
+   * isn't one with the same <route_config_name>_<cluster> already. Otherwise, it returns a
+   * Router::RouteConfigProviderSharedPtr created from the manager held weak_ptr.
+   * @param config supplies the json configuration of an RdsRouteConfigProvider.
+   * @param runtime supplies the runtime loader.
+   * @param cm supplies the ClusterManager.
+   * @param dispatcher supplies the dispatcher.
+   * @param random supplies the random generator.
+   * @param local_info supplies the local info.
+   * @param scope supplies the scope to use for the route config provider.
+   * @param stat_prefix supplies the stat_prefix to use for the provider stats.
+   * @param tls supplies the tls slot allocator.
+   * @param init_manager supplies the init manager.
+   */
+  virtual RouteConfigProviderSharedPtr
+  getRouteConfigProvider(const Json::Object& config, Runtime::Loader& runtime,
```

Comment:
Quick comment to work on (sorry really busy this week so I'm giving you drive by reviews :( ): From an interface perspective, I think almost all of these things don't need to be passed. They can be known internally to the route manager. I wouldn't pass any json either, and just unpack into exactly what you need which I think is just route table name and cluster name. (This will likely get converted to v2 but the conversion will be easy cc @htuch ). 

So I think at quick glance you need to pass cluster name, route table name, init manager, scope, and maybe stat_prefix.